### PR TITLE
[DNM] SIMD

### PR DESCRIPTION
**Pull request by stephentyrone**
Initial implementation of simd types and protocols for the standard library. Pending Swift-Evolution review.

### CODE REVIEW

**Comment by stephentyrone** [comment]
@swift-ci please test

**Comment by xedin** [comment]
@stephentyrone Quick question - would new types be loaded together with existing integer types or would they require additional `import`s? I'm asking because I see that they add a number of new operator overloads, so we'd need to consider what type-checker impact is going to be...

**Comment by rudkx** [comment]
@stephentyrone If `SIMDVector` is `ExpressibleByArrayLiteral` and `SIMDFloatingPointVector` provides `func +`, how is one (person or compiler) supposed to interpret `[1,1,1,1] + [1,1,1,1]`?

**Comment by stephentyrone** [comment]
@rudkx If I had my choice, we wouldn't interpret it, because using `+` for concatenation is a damned abomination whether or not literals are involved 😂.

... but that ship has sailed, and as a pragmatic matter I would prefer to interpret it as [Int] + [Int], since that's preserves the existing behavior, but that's not what currently falls out.

File: stdlib/public/core/SIMDInteger.swift [hunk 1]
```diff
@@ -0,0 +1,484 @@
...
+  @inlinable
+  static func > <Scalar>(lhs: Scalar, rhs: Self) -> Mask
+  where Scalar : BinaryInteger {
+    return rhs < lhs
+  }
+  
+  @inlinable
+  static func >= <Scalar>(lhs: Scalar, rhs: Self) -> Mask
+  where Scalar : BinaryInteger {
+    return rhs <= rhs
```
Comment (moiseev) [inline stdlib/public/core/SIMDInteger.swift]: One of the `rhs` should be `lhs`.

Comment (moiseev) [inline stdlib/public/core/SIMDInteger.swift]:
```suggestion
    return rhs <= lhs
```

**Pull request closed without merging**